### ISSUE TITLE
Out of Bounds Exception while editing a list

### ISSUE THREAD

**Issue by aomader** [stefan-niedermann/nextcloud-notes#1416]
**Describe/Reproduce the bug**

When creating a simple list (`- ...`), the app crashs when hitting enter on an empty point (`- <ENTER>`).

**Expected behavior**

It should not crash, but remove the empty point and create a new line.

**Smartphone:**

```
App Version: 3.4.16
App Version Code: 3004016
App Flavor: fdroid
Files App Version Code: 30180090
OS Version: 4.4.289-lineage+(10036820)
OS API Level: 30
Device: OnePlus5T
Manufacturer: OnePlus
Model (and Product): ONEPLUS A5010 (OnePlus5T)
```

**Stacktrace**

```
java.lang.IndexOutOfBoundsException: replace (95 ... 98) ends beyond length 97
 at 
android.text.SpannableStringBuilder.checkRange(SpannableStringBuilder.java:1325)
 at 
android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:513)
 at 
android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:507)
 at android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:37)
 at 
it.niedermann.android.markdown.markwon.textwatcher.AutoContinuationTextWatcher.deleteCustomText(AutoContinuationTextWatcher.java:78)
 at 
it.niedermann.android.markdown.markwon.textwatcher.AutoContinuationTextWatcher.afterTextChanged(AutoContinuationTextWatcher.java:60)
 at 
it.niedermann.android.markdown.markwon.textwatcher.LowerIndentionTextWatcher.afterTextChanged(LowerIndentionTextWatcher.java:49)
 at 
it.niedermann.android.markdown.markwon.textwatcher.SearchHighlightTextWatcher.afterTextChanged(SearchHighlightTextWatcher.java:61)
 at 
it.niedermann.android.markdown.markwon.textwatcher.CombinedTextWatcher.afterTextChanged(CombinedTextWatcher.java:47)
 at android.widget.TextView.sendAfterTextChanged(TextView.java:10681)
 at android.widget.TextView$ChangeWatcher.afterTextChanged(TextView.java:13668)
 at 
android.text.SpannableStringBuilder.sendAfterTextChanged(SpannableStringBuilder.java:1277)
 at 
android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:577)
 at 
android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:507)
 at android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:37)
 at android.text.method.QwertyKeyListener.onKeyDown(QwertyKeyListener.java:228)
 at android.text.method.TextKeyListener.onKeyDown(TextKeyListener.java:141)
 at android.widget.TextView.doKeyDown(TextView.java:8483)
 at android.widget.TextView.onKeyDown(TextView.java:8257)
 at android.view.KeyEvent.dispatch(KeyEvent.java:2846)
 at android.view.View.dispatchKeyEvent(View.java:14247)
 at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1959)
 at android.widget.ScrollView.dispatchKeyEvent(ScrollView.java:491)
 at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1959)
 at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1959)
 at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1959)
 at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1959)
 at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1959)
 at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1959)
 at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1959)
 at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1959)
 at 
com.android.internal.policy.DecorView.superDispatchKeyEvent(DecorView.java:476)
 at 
com.android.internal.policy.PhoneWindow.superDispatchKeyEvent(PhoneWindow.java:1863)
 at android.app.Activity.dispatchKeyEvent(Activity.java:4085)
 at 
androidx.core.app.ComponentActivity.superDispatchKeyEvent(ComponentActivity.java:122)
 at 
androidx.core.view.KeyEventDispatcher.dispatchKeyEvent(KeyEventDispatcher.java:84)
 at 
androidx.core.app.ComponentActivity.dispatchKeyEvent(ComponentActivity.java:140)
 at 
androidx.appcompat.app.AppCompatActivity.dispatchKeyEvent(AppCompatActivity.java:599)
 at 
androidx.appcompat.view.WindowCallbackWrapper.dispatchKeyEvent(WindowCallbackWrapper.java:59)
 at 
androidx.appcompat.app.AppCompatDelegateImpl$AppCompatWindowCallback.dispatchKeyEvent(AppCompatDelegateImpl.java:3068)
 at 
androidx.appcompat.view.WindowCallbackWrapper.dispatchKeyEvent(WindowCallbackWrapper.java:59)
 at com.android.internal.policy.DecorView.dispatchKeyEvent(DecorView.java:390)
 at 
android.view.ViewRootImpl$ViewPostImeInputStage.processKeyEvent(ViewRootImpl.java:5953)
 at 
android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:5821)
 at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:5316)
 at 
android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:5373)
 at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:5339)
 at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:5491)
 at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:5347)
 at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:5548)
 at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:5320)
 at 
android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:5373)
 at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:5339)
 at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:5347)
 at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:5320)
 at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:8086)
 at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:8037)
 at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:7998)
 at 
android.view.ViewRootImpl$ViewRootHandler.handleMessage(ViewRootImpl.java:5081)
 at android.os.Handler.dispatchMessage(Handler.java:106)
 at android.os.Looper.loop(Looper.java:223)
 at android.app.ActivityThread.main(ActivityThread.java:7664)
 at java.lang.reflect.Method.invoke(Native Method)
 at 
com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)

```

**Comment by stefan-niedermann** [comment]
Already fixed - will be part of `v3.6.17` :slightly_smiling_face: Thanks for the report though!

**Comment by aomader** [comment]
No problem, thank you for this awesome app!
